policy/policy.go

```diff
@@ -156,28 +241,29 @@ func NewPoliciesFromSchema(policies []*schema.Policy) ([]Policy, error) {
 // policies by resolution in ascending order, finest resolution first.
 // If two policies have the same resolution, the one with longer
 // retention period comes first.
-type ByResolutionAsc []Policy
+type ByResolutionAsc []UnaggregatedPolicy
 
 func (pr ByResolutionAsc) Len() int      { return len(pr) }
 func (pr ByResolutionAsc) Swap(i, j int) { pr[i], pr[j] = pr[j], pr[i] }
 
 func (pr ByResolutionAsc) Less(i, j int) bool {
-	w1, w2 := pr[i].Resolution().Window, pr[j].Resolution().Window
+	pi, pj := pr[i].Policy(), pr[j].Policy()
+	w1, w2 := pi.Resolution().Window, pj.Resolution().Window
 	if w1 < w2 {
 		return true
 	}
 	if w1 > w2 {
 		return false
 	}
-	r1, r2 := pr[i].Retention(), pr[j].Retention()
+	r1, r2 := pi.Retention(), pj.Retention()
 	if r1 > r2 {
 		return true
 	}
 	if r1 < r2 {
 		return false
 	}
 	// NB(xichen): compare precision to ensure a deterministic ordering.
-	return pr[i].Resolution().Precision < pr[i].Resolution().Precision
+	return pi.Resolution().Precision < pj.Resolution().Precision
```

Comment:
Do we also need to compare aggregation types to ensure deterministic ordering? Otherwise the ordering w.r.t. two unaggregated policies with the same policy can be nondeterministic.